Makefile

```diff
@@ -13,146 +12,102 @@ X86_64_ONLY=0
 
-.PHONY: build-wheelhouse
 
-define python_component
-# 1 Component Name - dash separated
 
-build-wheelhouse: ${ENV_DIR}/.$1-build-wheelhouse
-
-${ENV_DIR}/.$1-build-wheelhouse: \
-	$$(subst -,_,$1)_setup.py \
-	shared_setup.py \
-	common/setup.py \
-	$(shell find src/metaswitch -type f -not -name "*.pyc") \
-	$(shell find common/metaswitch -type f -not -name "*.pyc") \
-	src/metaswitch/clearwater/$$(subst mgr,manager,$$(subst -,_,$1))/alarm_constants.py \
-	$$(subst -,_,$1)-requirements.txt \
-	common/requirements-test.txt \
-	common/requirements.txt \
-	fv-requirements.txt \
-	shared-requirements.txt
-
-	rm -f $$@
-
-	# Ensure we have an up to date version of pip with wheel support
-	${PIP} install --upgrade pip==9.0.1
-	${PIP} install wheel==0.30.0
-
-	# Generate wheels
-	${PYTHON} $$(subst -,_,$1)_setup.py build -b build_$$(subst -,,$1) bdist_wheel -d $$(subst -,_,$1)_wheelhouse
-	${PYTHON} shared_setup.py build -b build_shared bdist_wheel -d $$(subst -,_,$1)_wheelhouse
-	cd common && WHEELHOUSE=../$$(subst -,_,$1)_wheelhouse make build_common_wheel
+${ENV_DIR}/.test_requirements: common/requirements-test.txt fv-requirements.txt ${ENV_DIR}/.wheels-installed
+	${PIP} install -r common/requirements-test.txt -r fv-requirements.txt
 
-	# Download the required dependencies
-	${PIP} wheel -w $$(subst -,_,$1)_wheelhouse -r $$(subst -,_,$1)-requirements.txt -r shared-requirements.txt -r common/requirements.txt --find-links $$(subst -,_,$1)_wheelhouse
+# Macro to define how the various etcd targets use python common
+define python_common_component
 
-	# Install the dependencies in the local environment for testing
-	${INSTALLER} --find-links $(subst -,_,$1)_wheelhouse -r $$(subst -,_,$1)-requirements.txt -r shared-requirements.txt -r common/requirements.txt
+# Add a target that builds the python-common wheel into the correct wheelhouse
+${ENV_DIR}/.$1_build_common_wheel: common/requirements.txt $(shell find common/metaswitch -type f -not -name "*.pyc")
+	cd common && WHEELHOUSE=../$1_wheelhouse make build_common_wheel
+	touch $$@
 
-	# Install test only requirements
-	${PIP} install -r common/requirements-test.txt -r fv-requirements.txt
+# Add dependency to the install-wheels to ensure we also install the python-common wheel
+${ENV_DIR}/.$1-install-wheels: ${ENV_DIR}/.$1_build_common_wheel
 
-	touch $$@
 endef
 
-$(eval $(call python_component,queue-mgr))
-$(eval $(call python_component,config-mgr))
-$(eval $(call python_component,cluster-mgr))
+# Queue manager definitions
+queue_mgr_SETUP = queue_mgr_setup.py shared_setup.py
+queue_mgr_REQUIREMENTS = queue_mgr-requirements.txt common/requirements.txt shared-requirements.txt
+queue_mgr_SOURCES = $(shell find src/metaswitch -type f -not -name "*.pyc") $(shell find common/metaswitch -type f -not -name "*.pyc")
+$(eval $(call python_common_component,queue_mgr))
+$(eval $(call python_component,queue_mgr))
```

Comment:
Should python_common_component not just call python_component as part of executing?